OneSignalSDK/onesignal/core/src/main/java/com/onesignal/internal/OneSignalImp.kt

```diff
@@ -120,16 +121,27 @@ internal class OneSignalImp : IOneSignal, IServiceProvider {
 
     // Services required by this class
     private var _user: IUserManager? = null
+        get() = services.getService()
```

Comment:
Do you think these should be `lazy`?, so getService() only have to be called once.

Reply:
We can certainly do that. I don't think we will gain much by doing this though as calling resolve() will just return the obj if it was already instantiated.

Reply:
ya maybe we don't add lazy, as `services.getService()` is already doing what `lazy` would be doing any way.